modules/ui/org.eclipse.fx.ui.controls/src/main/java/org/eclipse/fx/ui/controls/tree/TreeTableViewUtil.java

```diff
@@ -94,29 +97,39 @@
 	 * @return the view you passed for fluent API useage
 	 */
 	public static <S> TreeTableView<S> restoreState(TreeTableView<S> view, Memento m, Function<TreeTableColumn<S, ?>, String> columnKeyProvider, TreeTableView_Properties... properties) {
-		Map<String, TreeTableColumn<S, ?>> idToNodeMap = view.getColumns().stream().collect(Collectors.toMap(columnKeyProvider, c -> c));
+		Map<String, TreeTableColumn<S, ?>> idToNodeMap = view.getColumns().stream().filter(c -> columnKeyProvider.apply(c) != null).collect(Collectors.toMap(columnKeyProvider, c -> c));
 		Map<TreeTableColumn<S, ?>, String> nodeToIdMap = idToNodeMap.entrySet().stream().collect(Collectors.toMap(Entry::getValue, Entry::getKey));
 
-		for (TreeTableColumn<S, ?> c : view.getColumns()) {
-			String key = nodeToIdMap.get(c);
+		@SuppressWarnings({ "unchecked", "null" })
+		final @NonNull List<String> savedOrder = m.get("column_order", List.class, Collections.emptyList()); //$NON-NLS-1$
 
-			if (key == null) {
-				continue;
-			}
+		view.getColumns().addListener((ListChangeListener<TreeTableColumn<S, ?>>) change -> {
+			change.reset();

```

Comment:
why reset?

Reply:
workaround for jdk bug 8164096 (which is fixed in u122)